test/ignition.go

```diff
@@ -0,0 +1,82 @@
+package e2e
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/url"
+)
+
+func CreateIgnition(ignitionFile string, publicKey string, user string, password string) error {
+	var (
+		mode = 0644
+		root = "root"
+		yes  = true
+		no   = false
+	)
+
+	systemd := Systemd{
+		Units: []Unit{
+			{
+				Name:    "systemd-resolved.service",
+				Enabled: &no,
+				Mask:    &yes,
+			},
+		},
+	}
+
+	passwd := Passwd{
+		Users: []PasswdUser{
+			{
+				Name:         user,
+				PasswordHash: &password,
+				SSHAuthorizedKeys: []SSHAuthorizedKey{
+					SSHAuthorizedKey(publicKey),
+				},
+				Groups: []Group{
+					"wheel",
+					"sudo",
+				},
+			},
+		},
+	}
+
+	storage := Storage{
+		Files: []File{
+			{
+				Node: Node{
+					Group:     NodeGroup{Name: &root},
+					Path:      "/etc/resolv.conf",
+					User:      NodeUser{Name: &root},
+					Overwrite: &yes,
+				},
+				FileEmbedded1: FileEmbedded1{
+					Append: nil,
+					Contents: Resource{
+						Source: encodeData("nameserver 192.168.127.1"),
```

Comment:
With Fedora 34, NetworkManager used to do that directly. Maybe we can check a file or something else to be sure the upstream DNS is the one given by the DHCP?

Reply:
I'll take a look to see. I vaguely recall NM still being there, so maybe there is a setting i can flip to generate it as well.

Reply:
Ah actually it looks like NM is already overwriting what I write here so I can just change this to a delete.